doc/howto/autoinstall-validation.rst

```diff
@@ -0,0 +1,249 @@
+.. _autoinstall_validation:
+
+Autoinstall Validation
+=====================================
+
+The following how-to guide demonstrates how to perform pre-validation of a autoinstall config.
+
+Autoinstall config is validated against a :doc:`JSON schema <../reference/autoinstall-schema>` during runtime before it is applied. This check ensures existence of required keys and their data types, but does not guarantee the validity of the data provided (e.g., a bad :ref:`match directive <disk_selection_extensions>`). Additionally, the following validation script is unable to replicate some of the cloud-config based :ref:`delivery checks <how_the_delivery_is_verified>`. There are some basic checks performed to catch simple delivery-related errors, which you can read more about in the examples section, but the focus of the validation is on the Autoinstall configuration *after* it has been delivered to the installer.
+
+.. note::
+   See the cloud-init documentation for `how to validate your cloud-config`_.
+
+
+Pre-validating the Autoinstall configuration
+--------------------------------------------
+
+You can validate autoinstall config prior to install time by using the `validate-autoinstall-user-data script <https://github.com/canonical/subiquity/blob/main/scripts/validate-autoinstall-user-data.py>`_ in the Subiquity GitHub repository.
+
+Getting Started
+^^^^^^^^^^^^^^^
+
+Running the validation script requires downloading the Subiquity source code and installing the development dependencies. First, clone the Subiquity repository and ``cd`` into the root of the repository:
+
+.. code:: none
+
+   git clone git@github.com:canonical/subiquity.git && cd subiquity/
```

Comment:
I'd recommend cloning with the https protocol, rather than SSH ; so it works for everyone.